crates/services/p2p/src/p2p_service.rs

```diff
@@ -445,47 +459,96 @@ impl<Codec: NetworkCodec> FuelP2PService<Codec> {
                 }
             }
 
-            FuelBehaviourEvent::PeerInfo(peer_info_event) => match peer_info_event {
-                PeerInfoEvent::PeerIdentified { peer_id, addresses } => {
+            FuelBehaviourEvent::PeerReport(peer_info_event) => match peer_info_event {
```

Comment:
I don't think that would be good idea since `PeerReport` is not just related to `PeerManager` other behaviours might use it, for example `Discovery` behaviour that needs `addresses` for unrouteable peers.

Doing it that way would seem that PeerManager is the only part that is interested in the events.
Having said that, I see your point, too many things were exposed from `PeerManager` so I have refactored this part, exposed cleaner methods from `PeerManager` and now they handle events in which they are interested in.